version 8.2.2

mroz22 opened the pull request:
# 8.2.2

### Added
- Cardano: `getNativeScriptHash` call
- Cardano: support for 1854 and 1855 key paths
- Cardano: support for script addresses
- Cardano: support for token minting and burning
- Cardano: support for multi-sig transactions using native scripts

### Fixed:
- Ethereum:ethereumSignTransaction eip1559 support for T1

### Changed
- Updated dependencies.
- Fixed missing `hex` field in SignMessage and VerifyMessage types.
- Stellar: ensure type correctness; stricter types.
- Deps: general update of dependencies

___

ack @tsusanka

szymonlesisz committed:
chore: update webpack@5

szymonlesisz committed:
tests: karma production build tests

szymonlesisz committed:
ci: enable karma tests

gabrielKerekes committed:
feat(cardano): update protobuf messages for multisig

gabrielKerekes committed:
feat(cardano): update cardanoGetAddress call for multisig

gabrielKerekes committed:
feat(cardano): add CardanoGetNativeScriptHash call

gabrielKerekes committed:
feat(cardano): update cardanoSignTransaction call for multisig

gabrielKerekes committed:
chore(cardano): use enums in type tests

szymonlesisz committed:
chore: rename utils/hdnode > hdnodeUtils

and add unit tests for hdnodeUtils

szymonlesisz committed:
tests: move signtxVerify fixtures into separate file

szymonlesisz committed:
tests: add addressUtils unit tests

matejcik committed:
feat(stellar): strong typing in stellar protobuf

matejcik committed:
feat(stellar): disallow unspecified timebounds

dependabot[bot] committed:
chore(deps): bump nth-check from 2.0.0 to 2.0.1

Bumps [nth-check](https://github.com/fb55/nth-check) from 2.0.0 to 2.0.1.
- [Release notes](https://github.com/fb55/nth-check/releases)
- [Commits](https://github.com/fb55/nth-check/compare/v2.0.0...v2.0.1)

---
updated-dependencies:
- dependency-name: nth-check
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

szymonlesisz committed:
ci: split fetch and build stages. cache node_modules

szymonlesisz committed:
chore: remove babel-collect-imports dependency

szymonlesisz committed:
chore: bump minor/patch dependencies

szymonlesisz committed:
chore: update babel dependencies

szymonlesisz committed:
chore: update eslint and prettier dependencies

szymonlesisz committed:
docs: update dependencies docs

dependabot[bot] committed:
chore(deps): bump minimist from 1.2.0 to 1.2.5

Bumps [minimist](https://github.com/substack/minimist) from 1.2.0 to 1.2.5.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.0...1.2.5)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump glob-parent from 5.1.0 to 5.1.2

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.0 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.0...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump ws from 7.4.2 to 7.4.6

Bumps [ws](https://github.com/websockets/ws) from 7.4.2 to 7.4.6.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/7.4.2...7.4.6)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump path-parse from 1.0.6 to 1.0.7

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.6 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump tmpl from 1.0.4 to 1.0.5

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

szymonlesisz committed:
ci: fix missing artifacts in test stage

matejcik committed:
fix(stellar): ensure type correctness

marekrjpolak committed:
fix: add missing hex param to sign/verify types

szymonlesisz committed:
release v8.2.2-beta.1

gabrielKerekes committed:
chore(cardano): update docs and changelog

szymonlesisz committed:
tests: update ethereum fixtures via trezor-common

szymonlesisz committed:
tests: use trezor-common fixtures for stellarSignTransaction

mroz22 committed:
fix: cardano tx witness request message as obj

mroz22 committed:
fix: sharedConnectionWorker worker vs module behavior

szymonlesisz committed:
ci: enable composeTransaction tests

szymonlesisz committed:
chore: update utxo-lib@1.0.0

changes:
- "bchaddrjs" logic is now a part of utxo-lib.
- "hd-wallet/buildTx" logic is now a part of utxo-lib.
- reflecting changes is "utxo-lib/Network" type (added forkId, removed coin and dustThreshold)
- renamed utxo-lib(hd-wallet) types.
- reflecting changes is "utxo-lib/Transaction" (options param in fromHex method, better extra data handling)
- using "utxo-lib/bip32" instead of "HDNode". upstream of utxo-lib removed HDNode from their code a while ago. see: https://github.com/bitcoinjs/bitcoinjs-lib/blob/239711bf4ef00651af92049bcdf88b12252b945c/CHANGELOG.md#400
- using "utxo-lib/payments" module to derive addresses.
- removed unnecessary addressUtils, this logic moved to utxo-lib.

szymonlesisz committed:
tests: fix getAccountInfo fixture for xrp account

szymonlesisz committed:
fix: protobuf build script ignore unknown top level option

JoeGruffins committed:
feat: add Decred support

based on:
https://github.com/trezor/connect/pull/678

szymonlesisz committed:
tests: add Decred signTransaction and getAddress cases

mmilata committed:
fix: ethereumSignTransaction eip1559 support for T1

mroz22 committed:
chore: reflect changes from previous commit in tests

szymonlesisz committed:
feat: add Taproot in popup discovery mode

overcat committed:
chore: update trezor-common submodule.

overcat committed:
chore: update Stellar's protocol buffer related files.

overcat committed:
feat(stellar): add support for StellarManageBuyOfferOp and StellarPathPaymentStrictSendOp

mroz22 committed:
version 8.2.2

tsusanka reviewed (approved):
ACK from me. No breaking changes right?

mroz22 commented:
Personally, I am not aware of any breaking changes. But please let also @szymonlesisz ack this.

szymonlesisz commented:
ack, no breaking changes yet

szymonlesisz committed:
chore: update utxo-lib breaking changes

reflecting breaking changes in utxo/lib while developing from version beta.1 to beta.10

- composeTx module added "txType" params https://github.com/trezor/trezor-suite/pull/4421/commits/4bdec6c22a6d69bbfae3aad148fb2b56dfe6f704
- composeTx module removed "segwit" and "inputAmounts" params (same commit as above)
- renamed AccountType from normal/segwit/legacy > p2pkh/p2sh/p2wpkh
- composeTx test fixtures with lower fees https://github.com/trezor/trezor-suite/commit/fb9eb4b4f5a16367167db4fb383e1bbc93ff488e

szymonlesisz committed:
feat: add Taproot support

szymonlesisz committed:
release v8.2.3-beta.2
